### PR TITLE
ci: fixed workflow for generating introspection query for docs 

### PR DESCRIPTION
**Pull request by banasa44**
## Description

In this PR two problems are fixed:

1. We have now 2 processes that generate 2 different types of introspection-query for subgraph schema. Some modifications need to be done for the introspection for docs purposes. Create a temp schema, add some necessary lines for the introspection-partial, generate it, and delete the temp.schema.

https://subgraph.satsuma-prod.com/aragon/osx-carles-sepolia/version/2/playground

Task ID: [OS-1221](https://aragonassociation.atlassian.net/browse/OS-1221)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.
- [x] I have updated the Subgraph and added a QA URL to the description of this PR.

[OS-1221]: https://aragonassociation.atlassian.net/browse/OS-1221?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by josemarinas — APPROVED** [review]
Looks good

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,4 @@ schema-introspection.json
 
 # docs
 packages/subgraph/docs/developer-portal/
+schema-introspection-partial.json
```
Comment (Rekard0) [inline .gitignore:47]:
```suggestion
schema-introspection-partial.json

```

**Review by Rekard0 — COMMENTED** [review]
Could you deploy subgraph and provide QA url?

File: packages/subgraph/scripts/generate-introspection-json.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ graph create $SUBGRAPH_NAME-$NETWORK_NAME --node http://localhost:8020
 # deploy subgraph
 LOCAL=true ./scripts/deploy-subgraph.sh
 # make introspection json
-FILENAME="docs/schema-introspection.json"
+FILENAME="tests/utils/schema-introspection.json"
```
Comment (Michael-A-Heuer) [inline packages/subgraph/scripts/generate-introspection-json.sh:37]: Why this is located in the `tests/utils` folder? This seems wrong

Comment (banasa44) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: this introspection query is going to be used for testing the query examples

Comment (Rekard0) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: but it is also used for non-test related action ? if yes it should not be in tests/

File: .github/workflows/subgraph-documentation-update.yml [hunk 3]
```diff
@@ -28,8 +28,8 @@ jobs:
           node-version: 18
       - name: Install dependencies
         run: yarn install --frozen-lockfile
-      - name: Generate introspection query result
-        run: yarn generate:introspection
+      - name: Generate partial introspection query result for documentation
```
Comment (Michael-A-Heuer) [inline .github/workflows/subgraph-documentation-update.yml:31]:
```suggestion
      - name: Generate partial introspection query results for documentation
```

File: packages/subgraph/scripts/generate-introspection-json.sh (hunk 2 above)
Comment (banasa44) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: Not for the moment, and I can't see another use for us in the future.
Introspection queries are used to retrieve information about the GraphQL schema itself. Commonly used for auto-generating documentation, building dynamic queries, and performing schema validation.
We already have a different one for docs, and this one will only serve for schema validation (testing) purposes.

**Review by Michael-A-Heuer — DISMISSED** [review]

File: packages/subgraph/scripts/generate-introspection-json.sh (hunk 2 above)
Comment (Rekard0) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: Then the script `generate-introspection-json.sh` perhaps need to be moved to a scripts/tests/
WDYT?

**Review by Michael-A-Heuer — DISMISSED** [review]
LGTM. Please add a PR title that follows https://www.conventionalcommits.org/en/v1.0.0/
The PR titles will be visible in our release page by customers (see, for example, https://github.com/aragon/osx/releases/tag/v1.3.0).

File: packages/subgraph/scripts/generate-introspection-json.sh (hunk 2 above)
Comment (banasa44) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: makes sense

Comment (banasa44) [reply packages/subgraph/scripts/generate-introspection-json.sh:37]: @Rekard0 fixed in https://github.com/aragon/osx/pull/578/commits/ef3e3c864bed97ca2fd77be33e9d376b50324cc7

**Review by Rekard0 — APPROVED** [review]
LGTM!

**Review by josemarinas — APPROVED** [review]

**Pull request merged**